src/utils/tmpfile.ts

```diff
@@ -11,9 +11,10 @@ export const withTempFile = async <T>(
   handler: TempFileHandler<T>
 ): Promise<T> =>
   withTempDir(async (dir) => {
-    const file = await fs.open(path.join(dir, 'tempfile'), 'w+');
+    const filePath = path.join(dir, 'tempfile');
+    const file = await fs.open(filePath, 'w+');
     try {
-      return await handler(file);
+      return await handler(file, filePath);
```

Comment:
I wonder if this could be changed to just return the `filePath` as `tmpFile` is not used above in the fetcher?
```suggestion
      return await handler(filePath);
```